types/yargs/v10/index.d.ts

```diff
@@ -152,8 +152,10 @@ declare namespace yargs {
         options(key: string, options: Options): Argv;
         options(options: { [key: string]: Options }): Argv;
 
-        parse(): Arguments;
-        parse(arg: string | string[], context?: object, parseCallback?: ParseCallback): Arguments;
+        parse(arg?: string | ReadonlyArray<string>): Arguments;
+        parse(arg: string | ReadonlyArray<string>, parseCallback: ParseCallback): Arguments;
+        parse(arg: string | ReadonlyArray<string>, context: object): Arguments;
+        parse(arg: string | ReadonlyArray<string>, context: object, parseCallback: ParseCallback): Arguments;
```

Comment:
These two last signatures can be combined to
```ts
        parse(arg: string | ReadonlyArray<string>, context: object, parseCallback?: ParseCallback): Arguments;
```

Reply:
In my case the combination makes the `parseCallback()` argument fit into the `context` parameter for its `object` type, and `parseCallback` parameter become `undefined`, resulting in the type of parameters for `parseCallback()` not get inferred.